sample.py

```diff
@@ -0,0 +1,63 @@
+import torch
+from tqdm import tqdm
+
+from models.reverse_diffusion import Unet
+from models.forward_diffusion import ForwardDiffusion
+
+
+class Generator:
+    def __init__(self, model: Unet, forward_diffusion: ForwardDiffusion):
+        self.model = model.eval()
+        self.forward_diffusion = forward_diffusion.eval()
+        self.timesteps = forward_diffusion.timesteps
+
+    @torch.no_grad()
+    def sample(
+        self, x: torch.Tensor, timesteps: int = None, intermediate_outputs: bool = False
+    ) -> torch.Tensor | dict[torch.Tensor]:
+        """
+        Sample ECG signal
+
+        Args:
+            x (torch.Tensor): signal to denoise
+            timesteps (int, optional): timestep from which denoising starts if None denoising starts from forward_diffusion end timestep. Defaults to None.
+            intermediate_outputs (bool, optional): if True returns dict of intermediate outputs with timestep as key. Defaults to False.
```

Comment:
This is again about my personal convenience, but I work on a small monitor, and lines >120 characters long get broken and mess with visual indentation :)